src/display/canvas.js

```diff
@@ -1127,12 +1127,9 @@ function genericComposeSMask(
   const chunkSize = Math.min(height, Math.ceil(PIXELS_TO_PROCESS / width));
   for (let row = 0; row < height; row += chunkSize) {
     const chunkHeight = Math.min(chunkSize, height - row);
-    const maskData = maskCtx.getImageData(
-      layerOffsetX - maskOffsetX,
-      row + (layerOffsetY - maskOffsetY),
-      width,
-      chunkHeight
-    );
+    const maskX = Math.max(0, layerOffsetX - maskOffsetX);
```

Comment:
This PR should either get a test case, or if that's hard at the very least a comment here on why this is necessary to prevent it from being removed accidentally during e.g., refactoring. I think the line from your PR description, `In some PDF files the mask offset is larger than the layer offset. In this case first two parameters of getImageData become negative numbers and produce a memory leak.`, is already quite a good and descriptive one.

Reply:
It seems that this is *Node.js-specific*, which probably makes the need for a test-case slightly less important?
Also, the example provided below in *14.5 MB* in size (for a one-page document) so it's probably not a great reduced test-case unfortunately.

Reply:
Yep according to the `getImageData`doc:
https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/getImageData
it's ok to have negative `sx` and `sy` and the corresponding pixels in the extracted data should be black transparent.

Reply:
Should we perhaps view this as a bug in the [`node-canvas` package](https://github.com/Automattic/node-canvas) then, or do we want to consider accepting this work-around here?